File: 프로그래머스/unrated/138477. 명예의 전당 （1）/명예의 전당 （1）.java
import java.util.PriorityQueue;
import java.util.Queue;

class Solution {
    public int[] solution(int k, int[] score) {
        int[] answer = new int[score.length];
        Queue<Integer> queue = new PriorityQueue<>();
        if (k > score.length) {
            for (int i = 0; i < score.length; i++) {
                queue.add(score[i]);
                answer[i] = queue.peek();
            }
        } else {
            for (int i = 0; i < k; i++) {
                queue.add(score[i]);
                answer[i] = queue.peek();
            }
            
            for (int i = k; i < score.length; i++) {
                if (queue.peek() < score[i]) {
                    queue.add(score[i]);
                    queue.poll();
                }
                answer[i] = queue.peek();
            }
        }

        return answer;
    }
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.js
function solution(k, score) {
    const stack = [];
    const answer = [];
    score.forEach(val => {
        stack.push(val);
        stack.sort((a, b) => a - b);
        
        if(stack.length > k) stack.shift();
        
        answer.push(stack[0]);
    })
    return answer;
}